src/core/layout/qgslayoutitemattributetable.cpp

```diff
@@ -541,6 +551,9 @@ bool QgsLayoutItemAttributeTable::getTableContents( QgsLayoutTableContents &cont
     // We also need a list of just the cell contents, so that we can do a quick check for row uniqueness (when the
     // corresponding option is enabled)
     QVector< Cell > currentRow;
+#ifdef HAVE_SERVER_PYTHON_PLUGINS
+    mColumns = filteredColumns();
```

Comment:
If this is permanently replacing the previous column configuration with the filtered columns, won't it mean that columns stripped from one request with limited column visibility which comes before a second request with more permissive configuration are always missing from the second request?

Reply:
@nyalldawson of course the server is supposed to be stateless so we don't have a problem here. I've added a test case to prove it anyway.